### PR TITLE
 Fix: Add return objects for requests and correct return types in proto for invite service

### PR DESCRIPTION
**Pull request by skelet0no**
1. Added return objects for requests.
2. Corrected return types in proto for invite service.

### CODE REVIEW

**Review by MaxPlays35 — APPROVED** [review]

File: invite_service/repository/invite_repository_impl.py [hunk 1]
```diff
@@ -304,10 +309,9 @@ async def create_invite(self, invite: Invite) -> None:
         )
 
         if prisma_db_invite is None:
-            await self._db_client.db.invite.create(
+            return await self._db_client.db.invite.create(
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py]: wrong return type

File: invite_service/repository/invite_repository_impl.py [hunk 2]
```diff
@@ -316,9 +320,9 @@ async def create_invite(self, invite: Invite) -> None:
 
         db_invite.deleted_at = None
         db_invite.status = InviteStatus.PENDING
-        await self.update_invite(db_invite)
+        return await self.update_invite(db_invite)
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py]: wrong return type

File: invite_service/repository/invite_repository_impl.py [hunk 3]
```diff
@@ -351,14 +366,15 @@ async def create_multiple_invites(self, invites: List[Invite]) -> None:
...
                     "deleted_at": None,
                     "status": InviteStatus.PENDING,
-                }
+                },
             )
 
         await self._db_client.db.invite.create_many(
             data=[invite.to_dict() for invite in invites]
         )
+        return invites
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py]: wrong returning object

File: invite_service/repository/invite_repository_impl.py [hunk 4]
```diff
@@ -367,13 +383,18 @@ async def update_invite(self, invite: Invite) -> None:
...
+            Updated invite.
+
         Raises
         ------
         prisma.errors.PrismaError
             Catch all for every exception raised by Prisma Client Python.
 
         """
-        await self._db_client.db.invite.update(
+        return await self._db_client.db.invite.update(
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py]: wrong return type

**Review by ZixMai — APPROVED** [review]

**Review by ZixMai — CHANGES_REQUESTED** [review]
Make changes

File: invite_service/repository/mock_invite_repository.py [hunk 5]
```diff
@@ -289,15 +299,21 @@ async def create_invite(self, invite: Invite) -> None:
             invite.created_at = datetime.now()
             invite.deleted_at = None
```
Comment (ZixMai) [inline invite_service/repository/mock_invite_repository.py:300]: add line with invite.status = InviteStatus.PENDING

File: invite_service/repository/mock_invite_repository.py [hunk 6]
```diff
@@ -308,30 +324,49 @@ async def create_multiple_invites(self, invites: List[Invite]) -> None:
...
 
             self._invites[index].deleted_at = None
             self._invites[index].status = InviteStatus.PENDING
+            updated_invites.append(self._invites[index])
 
         ids = [db_invite.id for db_invite, _ in db_invites]
         invites = [invite for invite in invites if invite.id not in ids]
 
         for invite in invites:
+            invite.deleted_at = None
```
Comment (ZixMai) [inline invite_service/repository/mock_invite_repository.py:354]: add definition of created_at and id fields

**Review by ZixMai — CHANGES_REQUESTED** [review]

**Review by ZixMai — APPROVED** [review]

File: invite_service/repository/invite_repository_impl.py [hunk 7]
```diff
@@ -299,26 +304,31 @@ async def create_invite(self, invite: Invite) -> None:
...
+            if (
+                db_invite.status == InviteStatus.PENDING
+                and db_invite.deleted_at is None
+            ):
+                raise UniqueError("Invite already exists")
 
-    async def create_multiple_invites(self, invites: List[Invite]) -> None:
+            db_invite.deleted_at = None
+            db_invite.status = InviteStatus.PENDING
+            return Invite.from_prisma_invite(await self.update_invite(db_invite))
```
Comment (Nhsdkk) [inline invite_service/repository/invite_repository_impl.py]: Why are you calling another method in transaction. you should use transaction client here

File: invite_service/repository/invite_repository_impl.py [hunk 8]
```diff
@@ -367,14 +398,21 @@ async def update_invite(self, invite: Invite) -> None:
...
         ------
         prisma.errors.PrismaError
             Catch all for every exception raised by Prisma Client Python.
 
         """
-        await self._db_client.db.invite.update(
-            where={"id": invite.id}, data=invite.to_dict()
+        return Invite.from_prisma_invite(
+            await self._db_client.db.invite.update(
+                where={"id": invite.id}, data=invite.to_dict()
```
Comment (Nhsdkk) [inline invite_service/repository/invite_repository_impl.py:418]: deleted_at should be none

File: invite_service/repository/mock_invite_repository.py [hunk 9]
```diff
@@ -308,30 +325,51 @@ async def create_multiple_invites(self, invites: List[Invite]) -> None:
...
+        db_invites = [
+            (self._invites[i], i)
+            for i in range(len(self._invites))
+            if self._invites[i].id in invite_ids
+        ]
+        updated_invites = []
 
         for invite, index in db_invites:
-            if any([invite == InviteStatus.PENDING and invite.deleted_at is None for invite, _ in db_invites]):
+            if any(
```
Comment (Nhsdkk) [inline invite_service/repository/mock_invite_repository.py]: you should check this once

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

File: invite_service/repository/invite_repository_impl.py (hunk 8 above)
Comment (skelet0no) [reply invite_service/repository/invite_repository_impl.py:418]: Wont do it

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Pull request merged**